modules/apps/object/object-test/src/testFunctional/macros/ValidateObjectsUpgrade.macro

```diff
@@ -0,0 +1,92 @@
+definition {
+
+	macro addEmailNotificationTemplate {
+		Navigator.openURL();
+
+		ApplicationsMenu.gotoPortlet(
+			category = "Notifications",
+			panel = "Control Panel",
+			portlet = "Templates");
+
+		ObjectNotifications.addNewNotificationTemplate(
+			bcc = "${bcc}",
+			cc = "${cc}",
+			description = "${description}",
+			emailBody = "${emailBody}",
+			entryTo = "${entryTo}",
+			fromAddress = "${fromAddress}",
+			fromName = "${fromName}",
+			name = "${name}",
+			notificationType = "${notificationType}",
+			subject = "${subject}");
+
+		if (isSet(objectLabelAttachment)) {
+			ObjectNotifications.addNotificationTemplateAttachment(objectLabel = "${objectLabelAttachment}");
+		}
+
+		PortletEntry.save();
```

Comment:
Generally I don't recommend saving within the macro since it makes the macro a little inflexible. The idea here is that we can call a macro to add something and/or populate a page but allow the test to then make assertions or changes if needed. By saving it immediately, we need to go back to the page, which adds an additional step.